src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -95,26 +99,98 @@ private void setSubmissionData(Map<String, Object> data, String overseasEntityId
         data.put(MANAGING_OFFICERS_INDIVIDUAL_FIELD, submissionDto.getManagingOfficersIndividual());
         data.put(MANAGING_OFFICERS_CORPORATE_FIELD, submissionDto.getManagingOfficersCorporate());
         data.put(BENEFICIAL_OWNERS_STATEMENT, submissionDto.getBeneficialOwnersStatement());
 
+        ApiLogger.debug("Submission data has been set on filing", logMap);
+    }
+
+    private List<BeneficialOwnerIndividualDto> getBeneficialOwnersIndividualSubmissionData(OverseasEntitySubmissionDto submissionDto) throws ServiceException {
+        List<BeneficialOwnerIndividualDto> beneficialOwnersIndividualSubmissionData = new ArrayList<>();
+
+        if (Objects.isNull(submissionDto.getBeneficialOwnersIndividual())) {
+            return beneficialOwnersIndividualSubmissionData;
+        }
+
+        for (BeneficialOwnerIndividualDto beneficialOwner : submissionDto.getBeneficialOwnersIndividual()) {
+            String noTrustsMessage = "No trusts exist for this filing but a trust id is provided for BO Individual "
+                    + beneficialOwner.getFirstName() + " " + beneficialOwner.getLastName();
+
+            List<TrustDataDto> trustData = getTrustData(submissionDto, beneficialOwner.getTrustIds(), noTrustsMessage);
+            beneficialOwner.setTrustData(convertTrustDataToString(trustData));
+            beneficialOwnersIndividualSubmissionData.add(beneficialOwner);
+        }
+
+        return beneficialOwnersIndividualSubmissionData;
+    }
+
+    private List<BeneficialOwnerCorporateDto> getBeneficialOwnersCorporateSubmissionData(OverseasEntitySubmissionDto submissionDto) throws ServiceException {
+        List<BeneficialOwnerCorporateDto> beneficialOwnersCorporateSubmissionData = new ArrayList<>();
+
+        if (Objects.isNull(submissionDto.getBeneficialOwnersCorporate())) {
+            return beneficialOwnersCorporateSubmissionData;
+        }
+
+        for (BeneficialOwnerCorporateDto beneficialOwner : submissionDto.getBeneficialOwnersCorporate()) {
+            String noTrustsMessage = "No trusts exist for this filing but a trust id is provided for BO Corporate "
+                    + beneficialOwner.getPublicRegisterName();
+
+            List<TrustDataDto> trustData = getTrustData(submissionDto, beneficialOwner.getTrustIds(), noTrustsMessage);
+            beneficialOwner.setTrustData(convertTrustDataToString(trustData));
+            beneficialOwnersCorporateSubmissionData.add(beneficialOwner);
+        }
+
+        return beneficialOwnersCorporateSubmissionData;
+    }
+
+    private List<TrustDataDto> getTrustData(OverseasEntitySubmissionDto submissionDto, List<String> trustIds, String noTrustsExceptionMessage) throws ServiceException {
+        List<TrustDataDto> trustsDataForBO = new ArrayList<>();
+        // Loop through each trustId(s) and lookup the information for that trust
+        if (Objects.nonNull(trustIds)) {
```

Comment:
to reduce the cognitive complexity (see code smell) try ad ding a guard clause at top, replace line 53,

if (Objects.isNull(trustIds)) {
  return trustsDataForBO;
}

for (String trustId : trustIds) {
                List<TrustDataDto> trustData = new ArrayList<>();
etc....